### PR TITLE
Home page

### PR DESCRIPTION
**Pull request by itzhak-Israeli**

### CODE REVIEW

File: client/src/Components/ClickableBox.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import {replaceEngToHeb} from "../Utils/replaceEngToHeb";
+
+interface ClickableBoxProps {
+    categoryName: string,
+    subCategories: string[]
+}
+
+export const ClickableBox = ({categoryName, subCategories}: ClickableBoxProps) => {
+    const image = getImageByCategory(categoryName);
+    const title = replaceEngToHeb(categoryName);
```
Comment (AviranBac) [inline client/src/Components/ClickableBox.tsx]: After communicating with backend and getting the image urls, you won't need `getImageByCategory`, and the `categoryName` will already be in Hebrew, so `replaceEngToHeb` will be removed as well.

File: client/src/Screens/HomeScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import {Image, ScrollView, Text, TouchableOpacity, View} from "react-native";
+import React from "react";
+import {ClickableBox} from "../Components/ClickableBox";
+
+const CATEGORIES_LIST: [string, string[]][] = [['play', []], ['dress', []]];
```
Comment (AviranBac) [inline client/src/Screens/HomeScreen.tsx]: As we spoke, make something like a service that will be responsible for fetching the categories from the backend (using `axios`).  The response will only include categories (without subcategories).

File: client/src/ButtomBar/BottomBar.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+import {HomeScreen} from "../Screens/HomeScreen";
+import {FavoritesScreen} from "../Screens/FavoritesScreen";
+import {RecentScreen} from "../Screens/RecentsScreen";
+import {SettingsScreen} from "../Screens/SettingsScreen";
+
+const Tab = createMaterialBottomTabNavigator();
+
+const TABS = {
+    HOME: "בית",
+    RECENT: 'בשימוש נפוץ',
```
Comment (AviranBac) [inline client/src/ButtomBar/BottomBar.tsx]: `RecentScreen` -> `CommonlyUsedScreen`, rename the relevant screen

File: client/src/Components/ClickableBox.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    return (
+        <View style={{flexBasis: '50%', alignItems:'center', justifyContent:'center'}}>
```
Comment (AviranBac) [inline client/src/Components/ClickableBox.tsx]: Move `flexBasis`, `alignItems`, `justifyContent` to the parent (`HomePage`)
You hard-codedly duplicated 6 items per category here but in reality there'll be only one `TouchableOpacity` in here, without a `View`

Also, according to the mockup, there should be 4 items in each row

File: client/src/Components/ClickableBox.tsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+            <TouchableOpacity>
```
Comment (AviranBac) [inline client/src/Components/ClickableBox.tsx:15]: In general, try to make the `TouchableOpacity` look like the mockup we originally did:
border for the picture, text under it.. whatever you can
In the end of the project there's a section in which we need to make a comparison between the mockup and the end-product, so it's important we stay consistent as much as possible :)

File: client/package.json [hunk 6]
```diff
@@ -9,11 +9,15 @@
     "web": "expo start --web"
```
Comment (AviranBac) [inline client/package.json:9]: In `app.json`:
Change `"orientation": "portrait"` to `"orientation": "default"`.
Now you can check how the app looks in landscape display whenever you flip your device.

**Review by AviranBac — CHANGES_REQUESTED** [review]

File: client/src/AppBar/MainAppBar.tsx [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import React from "react";
+import {AppBar} from "@react-native-material/core";
+import {Platform} from "react-native";
+import {Image} from "react-native";
+
+const logo = require('../../assets/logo-spic.png');
+export const MainAppBar = () => {
+    return (
+        <AppBar style={{direction: "rtl", backgroundColor: '#c7f5fe', paddingTop: (Platform.OS === 'ios') ? 20 : 0}}>
```
[the remark this answers is not in the document]
Comment (itzhak-Israeli) [reply client/src/AppBar/MainAppBar.tsx:9]: for now its ok to leave it like this
I saw this in SO

File: client/package.json [hunk 8]
```diff
@@ -9,11 +9,15 @@
...
   },
   "dependencies": {
+    "@react-native-material/core": "^1.3.7",
+    "@react-navigation/material-bottom-tabs": "^6.2.15",
     "expo": "~47.0.12",
     "expo-status-bar": "~1.4.2",
     "react": "^18.1.0",
     "react-dom": "^18.1.0",
     "react-native": "^0.70.5",
+    "react-native-paper": "^5.2.0",
```
[the remark this answers is not in the document]
Comment (itzhak-Israeli) [reply client/package.json:19]: They are currently not in use, but I think that We will need them later

File: client/src/ButtomBar/BottomBar.tsx [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+    FAVORITES: 'מועדפים',
+    SETTINGS: 'הגדרות משתמש'
+}
+
+export const BottomBar = () => {
+    return (
+        <NavigationContainer>
+            <Tab.Navigator
```
[the remark this answers is not in the document]
Comment (itzhak-Israeli) [reply client/src/ButtomBar/BottomBar.tsx:22]: fix this - add the rtl to the main page

File: client/src/Components/ClickableBox.tsx [hunk 10]
```diff
@@ -0,0 +1,41 @@
+import {Button, Image, TextBase, TouchableOpacity, View} from "react-native";
+import {getImageByCategory} from "../Utils/getImageByCategory";
+import {replaceEngToHeb} from "../Utils/replaceEngToHeb";
+
+interface ClickableBoxProps {
+    categoryName: string,
+    subCategories: string[]
+}
+
+export const ClickableBox = ({categoryName, subCategories}: ClickableBoxProps) => {
```
[the remark this answers is not in the document]
Comment (itzhak-Israeli) [reply client/src/Components/ClickableBox.tsx:10]: fix

File: client/src/Components/ClickableBox.tsx (hunk 1 above)
Comment (itzhak-Israeli) [reply client/src/Components/ClickableBox.tsx]: ok

File: client/src/Screens/HomeScreen.tsx (hunk 2 above)
Comment (itzhak-Israeli) [reply client/src/Screens/HomeScreen.tsx]: ok will be fixed when the back will be ready

File: client/src/AppBar/MainAppBar.tsx [hunk 11]
```diff
@@ -0,0 +1,15 @@
+import React from "react";
+import {AppBar} from "@react-native-material/core";
+import {Platform} from "react-native";
+import {Image} from "react-native";
+
+const logo = require('../../assets/logo-spic.png');
+export const MainAppBar = () => {
```
[the remark this answers is not in the document]
Comment (itzhak-Israeli) [reply client/src/AppBar/MainAppBar.tsx:7]: ok

File: client/src/Screens/HomeScreen.tsx [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+
+export const HomeScreen = () => {
+    return (
+        <ScrollView>
+            <View style={{flex: 1, justifyContent: 'center', alignItems: 'center', paddingTop: 10}}>
+                <View style={{flexGrow: 0, flexShrink: 0, flexBasis: '15%'}}>
+                    <Text style={{fontSize: 25}}>
```
[the remark this answers is not in the document]
Comment (itzhak-Israeli) [reply client/src/Screens/HomeScreen.tsx:12]: fix and add styled components

**Review by AviranBac — APPROVED** [review]

**Pull request merged**